src/components/structures/TimelinePanel.tsx

```diff
@@ -627,10 +630,18 @@ class TimelinePanel extends React.Component<IProps, IState> {
         });
     };
 
+    private onRoomTimelineRefresh = (room: Room, timelineSet: EventTimelineSet): void => {
+        console.log(`onRoomTimelineRefresh skipping=${timelineSet !== this.props.timelineSet}`);
+        if (timelineSet !== this.props.timelineSet) return;
+
+        this.refreshTimeline();
+    };
+
     private onRoomTimelineReset = (room: Room, timelineSet: EventTimelineSet): void => {
+        console.log(`onRoomTimelineReset skipping=${timelineSet !== this.props.timelineSet} skippingBecauseAtBottom=${this.canResetTimeline()}`);
         if (timelineSet !== this.props.timelineSet) return;
 
-        if (this.messagePanel.current && this.messagePanel.current.isAtBottom()) {
+        if (this.canResetTimeline()) {
```

Comment:
This seems like a missed refactor. `canResetTimeline` by name seems aplicable here and the logic appears pretty equivalent.

https://github.com/matrix-org/matrix-react-sdk/blob/27118a97993ea267eb6db398f7098ce61c5ff4b0/src/components/structures/RoomView.tsx#L1004-L1009

https://github.com/matrix-org/matrix-react-sdk/blob/27118a97993ea267eb6db398f7098ce61c5ff4b0/src/components/structures/TimelinePanel.tsx#L638